src/main/java/com/aws/greengrass/clientdevices/auth/iot/IotAuthClient.java

```diff
@@ -122,12 +141,67 @@ public boolean isThingAttachedToCertificate(Thing thing, Certificate certificate
             } catch (Exception e) {
                 logger.atError().cause(e).kv("thingName", thing.getThingName())
                         .kv("certificateId", certificate.getCertificateId())
-                        .log("Failed to verify certificate thing association. Check that the core device's IoT policy"
-                                + " grants the greengrass:VerifyClientDeviceIoTCertificateAssociation permission.");
+                        .log("Failed to verify certificate thing association. Check that the core device's "
+                                + "IoT policy grants the greengrass:VerifyClientDeviceIoTCertificateAssociation "
+                                + "permission.");
                 throw new CloudServiceInteractionException(
                         String.format("Failed to verify certificate %s thing %s association",
                                 certificate.getCertificateId(), thing.getThingName()), e);
             }
         }
+
+        @Override
+        @SuppressWarnings("PMD.AvoidCatchingGenericException")
+        public Stream<Thing> getThingsAssociatedWithCoreDevice() {
+            DeviceConfiguration configuration = clientFactory.getDeviceConfiguration();
+            String thingName = Coerce.toString(configuration.getThingName());
+
+            ListClientDevicesAssociatedWithCoreDeviceRequest request =
+                    ListClientDevicesAssociatedWithCoreDeviceRequest.builder()
+                            .coreDeviceThingName(thingName)
+                            .build();
+
+            try (GreengrassV2Client client = getGGV2Client(configuration)) {
+                ListClientDevicesAssociatedWithCoreDeviceResponse response =
+                        client.listClientDevicesAssociatedWithCoreDevice(request);
+
+                return response.associatedClientDevices().stream()
+                        .map(AssociatedClientDevice::thingName)
+                        .map(Thing::of);
+            }  catch (SdkClientException | AwsServiceException e) {
+                throw new CloudServiceInteractionException("Failed to list things associated with core", e);
+            }
+        }
+
+        // TODO: This should not live here ideally it should be returned by the clientFactory but we
+        //  are adding it here to avoid introducing new changes to the nucleus
+        private GreengrassV2Client getGGV2Client(DeviceConfiguration deviceConfiguration) {
+            ApacheHttpClient.Builder httpClient = ClientConfigurationUtils
+                    .getConfiguredClientBuilder(deviceConfiguration);
+            GreengrassV2ClientBuilder clientBuilder = GreengrassV2Client.builder()
+                    .credentialsProvider(AnonymousCredentialsProvider.create())
```

Comment:
@MikeDombo may have comments on if we're doing this correctly

Reply:
@MikeDombo could you check if this looks good? - If not what other way do we have to get a GreengrassV2Client?

Reply:
What's the purpose of this change? You already have access to a client from the Nucleus itself so that you don't need to handle this on your own.

Yes, this all looks fine otherwise, especially if you've taken it from the Nucleus.

Reply:
The main reason to do this is that the Nucleus provides a GGV2DataClient which provides access to data plain apis but it does not provide a GGV2Client to access the control plain api that has the operation we want to run.

The reason we need to call that API is to know what is still attached to the core vs what has been detached to update the local registries

Reply:
`RegionUtils.getGreengrassControlPlaneEndpoint(awsRegion, this.envStage)` should make getting the endpoint simpler.

Reply:
https://github.com/aws-greengrass/aws-greengrass-nucleus/blob/dd8e059e802f44a763819a01e2088263ae56f3b3/src/main/java/com/aws/greengrass/easysetup/DeviceProvisioningHelper.java#L160